hal/api/PwmOut.h

```diff
@@ -67,6 +67,10 @@ class PwmOut {
         core_util_critical_section_exit();
     }
 
+    /** To Free up pwm pin.
+     */
+    virtual ~PwmOut();
```

Comment:
Seconded.

Also the dtor can be inlined here, removing the need to add `PwmOut.cpp`. Since the function is just a single call to the c api there is no benefit to declaring the function externally.

Looking at this file, should pwmout_free be in a critical section?